Thanks Victor for writing this PR!

I've tested it locally and it seems there's something wrong:
```python
./python.exe -m test --junit-xml xml.out -R 3:3 -j 8
... Many lines
Traceback (most recent call last):
  File "/Users/admin/Projects/cpython/Lib/runpy.py", line 198, in _run_module_as_main
    return _run_code(code, main_globals, None,
                     "__main__", mod_spec)
  File "/Users/admin/Projects/cpython/Lib/runpy.py", line 88, in _run_code
    exec(code, run_globals)
    ~~~~^^^^^^^^^^^^^^^^^^^
  File "/Users/admin/Projects/cpython/Lib/test/__main__.py", line 2, in <module>
    main(_add_python_opts=True)
    ~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/admin/Projects/cpython/Lib/test/libregrtest/main.py", line 739, in main
    Regrtest(ns, _add_python_opts=_add_python_opts).main(tests=tests)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/Users/admin/Projects/cpython/Lib/test/libregrtest/main.py", line 731, in main
    exitcode = self.run_tests(selected, tests)
  File "/Users/admin/Projects/cpython/Lib/test/libregrtest/main.py", line 568, in run_tests
    return self._run_tests(selected, tests)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/Users/admin/Projects/cpython/Lib/test/libregrtest/main.py", line 528, in _run_tests
    self._run_tests_mp(runtests, self.num_workers)
    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/admin/Projects/cpython/Lib/test/libregrtest/main.py", line 426, in _run_tests_mp
    RunWorkers(num_workers, runtests, self.logger, self.results).run()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/admin/Projects/cpython/Lib/test/libregrtest/run_workers.py", line 600, in run
    result = self._process_result(item)
  File "/Users/admin/Projects/cpython/Lib/test/libregrtest/run_workers.py", line 571, in _process_result
    self.results.accumulate_result(result, self.runtests)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/admin/Projects/cpython/Lib/test/libregrtest/results.py", line 132, in accumulate_result
    self.add_junit(xml_data)
    ~~~~~~~~~~~~~~^^^^^^^^^^
  File "/Users/admin/Projects/cpython/Lib/test/libregrtest/results.py", line 165, in add_junit
    self.testsuite_xml.append(ET.fromstring(e))
                              ~~~~~~~~~~~~~^^^
  File "/Users/admin/Projects/cpython/Lib/xml/etree/ElementTree.py", line 1343, in XML
    return parser.close()
           ~~~~~~~~~~~~^^
xml.etree.ElementTree.ParseError: no element found: line 1, column 5239
```

--- a/Lib/test/libregrtest/refleak.py
+++ b/Lib/test/libregrtest/refleak.py
@@ -28,23 +28,21 @@ def save_support_xml(filename):
     if support.junit_xml_list is None:
         return
 
-    with open(filename, 'x', encoding='ascii') as fp:
-        for elem in support.junit_xml_list:
-            print(elem, file=fp)
+    import pickle
+    with open(filename, 'xb') as fp:
+        pickle.dump(support.junit_xml_list, fp)
     support.junit_xml_list = None
 
 
 def restore_support_xml(filename):
     try:
-        fp = open(filename, 'r', encoding='ascii')
+        fp = open(filename, 'rb')
     except FileNotFoundError:
         return
 
-    xml_list = []
+    import pickle
     with fp:
-        for line in fp:
-            line = line.rstrip()
-            xml_list.append(line)
+        xml_list = pickle.load(fp)
     os.unlink(filename)
 
     support.junit_xml_list = xml_list

